lua/lvim/core/autocmds.lua

```diff
@@ -180,16 +180,23 @@ function M.toggle_format_on_save()
 end
 
 function M.enable_reload_config_on_save()
-  local user_config_file = require("lvim.config"):get_user_config_path()
+  local pattern = get_config_dir() .. "/*.lua"
```

Comment:
It's better to avoid globals whenever possible, it makes the code easier to reason with and refactoring easier

```suggestion
  local user_config_file = require("lvim.config"):get_user_config_path()
  -- autocmds require forward slashes (even on windows)
  local pattern = user_config_file:gsub("\\", "/") .. "/*.lua"
```